### PR TITLE
avoid SVG 1.2 compositing operators

### PR DESCRIPTION
**Pull request by mjg**
Note: I have not checked the effect of this on printing yet. It helps with the issue mentioned in #2575, though.

xournalpp paints a white background behind the PDF even for export (but
nor for printing). It does so by painting it after the PDF is rendered,
specifying CAIRO_OPERATOR_DEST_OVER as the compositing operator to get
it below the PDF rendering. When rendering to a SVG 1.2 surface cairo
turns this into a comp-op:dst-over style.

So what's the problem? SVG 1.2 never got accepted as a standard, and
thus it is supported only partially. The resulting SVG does not even
work in inkscape.

Instead, render the white background first and avoid the compositing
operator. This dies not change the resulting image for png outputs but
helps to produce simpler svg.

### CODE REVIEW

**Comment by Febbe** [comment]
This may produce merge conflicts with #2435, so we must wait until this is merged

**Comment by mjg** [comment]
> This may produce merge conflicts with #2435, so we must wait until this is merged

They do not touch a common file at all, so there is no technical merge conflict, but there might be a functional one. On the other hand, if this PR here (or another fix) gets merged I would prefer to adapt the commit messages in #2575 before that gets merged.

**Comment by personalizedrefrigerator** [comment]
> This may produce merge conflicts with #2435, so we must wait until this is merged

I'm willing to fix merge conflicts!

**Review by rolandlo — APPROVED** [review]
I have tested it successfully. Since @personalizedrefrigerator is willing to handle potential merge conflicts, I will merge it.

**Comment by rolandlo** [comment]
@mjg Thanks for the contribution!

**Pull request merged**
